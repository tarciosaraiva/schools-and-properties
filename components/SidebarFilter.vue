<template>
  <div class="container">
    <h2>Filter</h2>
    <form>
      <fieldset>
        <legend>Property options</legend>
        <div class="row">
          <label class="column label-inline">
            <input id="p_house_type" v-model="pTypes" type="checkbox" value="House"> House
          </label>
          <label class="column label-inline">
            <input id="p_townhouse_type" v-model="pTypes" type="checkbox" value="Townhouse"> Townhouse
          </label>
          <label class="column label-inline">
            <input id="p_unit_type" v-model="pTypes" type="checkbox" value="ApartmentUnitFlat"> Unit
          </label>
        </div>
        <label for="p_rooms">Rooms</label>
        <select id="p_rooms" v-model="pRooms">
          <option value="1">1+</option>
          <option value="2">2+</option>
          <option value="3">3+</option>
          <option value="4">4+</option>
        </select>
        <label for="p_bathrooms">Bathrooms</label>
        <select id="p_bathrooms" v-model="pBathrooms">
          <option value="1">1+</option>
          <option value="2">2+</option>
          <option value="3">3+</option>
          <option value="4">4+</option>
        </select>
        <label for="p_car_spaces">Car spaces</label>
        <select id="p_car_spaces" v-model="pCarSpaces">
          <option value="1">1+</option>
          <option value="2">2+</option>
          <option value="3">3+</option>
          <option value="4">4+</option>
        </select>
        <label for="p_max_price">Max price</label>
        <select id="p_max_price" v-model="pMaxPrice">
          <option value="500000">500 K</option>
          <option value="600000">600 K</option>
          <option value="700000">700 K</option>
          <option value="800000">800 K</option>
          <option value="900000">900 K</option>
          <option value="1000000">1 M</option>
        </select>
      </fieldset>
      <button @click.stop.prevent="applyFilter">Find matching properties</button>
      <fieldset>
        <legend>School options</legend>
        <div class="row">
          <label class="column label-inline">
            <input id="s_edu_sector" v-model="sEducationSector" type="radio" name="s_edu_sector" value="all"> All
          </label>
          <label class="column label-inline">
            <input id="s_edu_sector" v-model="sEducationSector" type="radio" name="s_edu_sector" value="Government"> Public
          </label>
          <label class="column label-inline">
            <input id="s_edu_sector" v-model="sEducationSector" type="radio" name="s_edu_sector" value="NonGovernment"> Private
          </label>
        </div>
        <label for="s_rating">Overall rating</label>
        <select id="s_rating" v-model="sRating">
          <option value="100">= 100</option>
          <option value="99">>= 99</option>
          <option value="98">>= 98</option>
          <option value="97">>= 97</option>
          <option value="96">>= 96</option>
          <option value="95">>= 95</option>
          <option value="94">>= 94</option>
          <option value="93">>= 93</option>
          <option value="92">>= 92</option>
          <option value="91">>= 91</option>
          <option value="90">>= 90</option>
        </select>
        <div class="row">
          <label class="column" for="s_english_rating">English rating</label>
          <label class="column" for="s_maths_rating">Maths rating</label>
        </div>
        <div class="row">
          <div class="column">
            <select id="s_english_rating" v-model="sEnglishRating">
              <option value="5">= 5</option>
              <option value="4">>= 4</option>
              <option value="3">>= 3</option>
              <option value="2">>= 2</option>
              <option value="1">>= 1</option>
            </select>
          </div>
          <div class="column">
            <select id="s_maths_rating" v-model="sMathsRating">
              <option value="5">= 5</option>
              <option value="4">>= 4</option>
              <option value="3">>= 3</option>
              <option value="2">>= 2</option>
              <option value="1">>= 1</option>
            </select>
          </div>
        </div>
      </fieldset>
    </form>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { mapState } from 'vuex'

export default Vue.extend({
  props: {
    loadListingsFn: Function
  },
  computed: {
    pTypes: {
      get () { return this.filter.properties.types },
      set (value) { this.$store.commit('UPDATE_PROPERTY_FILTER_TYPE', value) }
    },
    pRooms: {
      get () { return this.filter.properties.rooms },
      set (value) { this.$store.commit('UPDATE_PROPERTY_FILTER_ROOMS', value) }
    },
    pBathrooms: {
      get () { return this.filter.properties.bathrooms },
      set (value) { this.$store.commit('UPDATE_PROPERTY_FILTER_BATHROOMS', value) }
    },
    pCarSpaces: {
      get () { return this.filter.properties.carSpaces },
      set (value) { this.$store.commit('UPDATE_PROPERTY_FILTER_CAR_SPACES', value) }
    },
    pMaxPrice: {
      get () { return this.filter.properties.maxPrice },
      set (value) { this.$store.commit('UPDATE_PROPERTY_FILTER_MAX_PRICE', value) }
    },
    sEducationSector: {
      get () { return this.filter.schools.educationSector },
      set (value) { this.$store.commit('UPDATE_SCHOOL_EDUCATION_SECTOR', value) }
    },
    sRating: {
      get () { return this.filter.schools.rating },
      set (value) { this.$store.commit('UPDATE_SCHOOL_RATING', value) }
    },
    sEnglishRating: {
      get () { return this.filter.schools.englishRating },
      set (value) { this.$store.commit('UPDATE_SCHOOL_ENG_RATING', value) }
    },
    sMathsRating: {
      get () { return this.filter.schools.mathsRating },
      set (value) { this.$store.commit('UPDATE_SCHOOL_MATH_RATING', value) }
    },
    ...mapState({ filter: (state: any) => state.filter })
  },
  methods: {
    applyFilter () {
      this.loadListingsFn()
    }
  }
})
</script>
