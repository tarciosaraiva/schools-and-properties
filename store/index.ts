import { GetterTree, ActionTree, MutationTree } from 'vuex'

interface ApplicationFilter {
  properties: PropertiesFilter,
  schools: SchoolsFilter
}

interface PropertiesFilter {
  types: string[],
  rooms: number,
  bathrooms: number,
  carSpaces: number,
  maxPrice: number
}

export interface SchoolsFilter {
  educationSector: string,
  zone: string,
  primary: SchoolLevelRating,
  secondary: SchoolLevelRating,
}

interface SchoolLevelRating {
  plot: boolean,
  rating: number,
  englishRating: number,
  mathsRating: number
}

export interface PropertyListing {
  id: number,
  price: string,
  area: string
  bathrooms: number,
  bedrooms: number,
  carspaces: number,
  displayableAddress: string,
  features: string[],
  landArea: number,
  latitude: number,
  longitude: number,
  postcode: string,
  propertyType: string
  region: string,
  state: string,
  street: string,
  streetNumber: string,
  suburb: string,
  unitNumber: string,
  media: string[],
  listingSlug: string
}

export interface School {
  schoolName: string,
  educationSector: string,
  schoolType: string,
  phoneNumber: string,
  primaryOverallScore: string,
  primaryEnglishScore: string,
  primaryMathsScore: string,
  primaryEnrolments: string,
  secondaryOverallScore: string,
  secondaryEnglishScore: string,
  secondaryMathsScore: string,
  secondaryEnrolments: string
}

export const state = () => ({
  bounds: {
    ne: { lat: 0, lon: 0 },
    nw: { lat: 0, lon: 0 },
    se: { lat: 0, lon: 0 },
    sw: { lat: 0, lon: 0 },
  },
  geocodeSuggestions: [] as any,
  list: [] as PropertyListing[],
  filter: {
    properties: {
      types: ['House', 'Townhouse'],
      rooms: 3,
      bathrooms: 2,
      carSpaces: 1,
      maxPrice: 900000
    },
    schools: {
      educationSector: 'all',
      zone: 'none',
      primary: {
        plot: true,
        rating: 95,
        englishRating: 4,
        mathsRating: 4
      },
      secondary: {
        plot: true,
        rating: 95,
        englishRating: 4,
        mathsRating: 4
      }
    }
  } as ApplicationFilter
})

export type RootState = ReturnType<typeof state>

export const getters: GetterTree<RootState, RootState> = {
  listings: state => state.list,
  schoolsFilter: state => state.filter.schools,
  propertiesFilter: state => state.filter.properties,
  locations: state => state.geocodeSuggestions,
  listingPois: state => state.list.map(l => ({
    type: 'Feature',
    id: l.id,
    geometry: {
      type: 'Point',
      coordinates: [l.longitude, l.latitude]
    },
    properties: { ...l }
  }))
}

export const mutations: MutationTree<RootState> = {
  ADD_LISTINGS (state, listings: any[]) {
    const propertyListings = listings
      .filter(l => l.type === 'PropertyListing')
      .map(({ listing }) => {
        const propertyListing: PropertyListing = {
          id: listing.id,
          price: listing.priceDetails?.displayPrice,
          area: listing.propertyDetails.area,
          bathrooms: listing.propertyDetails.bathrooms,
          bedrooms: listing.propertyDetails.bedrooms,
          carspaces: listing.propertyDetails.carspaces,
          displayableAddress: listing.propertyDetails.displayableAddress,
          features: listing.propertyDetails.features,
          landArea: listing.propertyDetails.landArea,
          latitude: listing.propertyDetails.latitude,
          longitude: listing.propertyDetails.longitude,
          postcode: listing.propertyDetails.postcode,
          propertyType: listing.propertyDetails.propertyType,
          region: listing.propertyDetails.region,
          state: listing.propertyDetails.state,
          street: listing.propertyDetails.street,
          streetNumber: listing.propertyDetails.streetNumber,
          suburb: listing.propertyDetails.suburb,
          unitNumber: listing.propertyDetails.unitNumber,
          media: listing.media.map((m: any) => m.url),
          listingSlug: listing.listingSlug
        }

        return propertyListing
      })

    state.list = propertyListings
  },
  UPDATE_PROPERTY_FILTER_TYPE (state, payload) {
    state.filter.properties.types = payload
  },
  UPDATE_PROPERTY_FILTER_ROOMS (state, payload) {
    state.filter.properties.rooms = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_BATHROOMS (state, payload) {
    state.filter.properties.bathrooms = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_CAR_SPACES (state, payload) {
    state.filter.properties.carSpaces = Number(payload)
  },
  UPDATE_PROPERTY_FILTER_MAX_PRICE (state, payload) {
    state.filter.properties.maxPrice = Number(payload)
  },
  UPDATE_BOUNDING_BOX (state, payload) {
    state.bounds.nw.lat = payload.getNorthWest().lat
    state.bounds.nw.lon = payload.getNorthWest().lng
    state.bounds.ne.lat = payload.getNorthEast().lat
    state.bounds.ne.lon = payload.getNorthEast().lng
    state.bounds.se.lat = payload.getSouthEast().lat
    state.bounds.se.lon = payload.getSouthEast().lng
    state.bounds.sw.lat = payload.getSouthWest().lat
    state.bounds.sw.lon = payload.getSouthWest().lng
  },
  UPDATE_SCHOOL_EDUCATION_SECTOR (state, payload) {
    state.filter.schools.educationSector = payload
  },
  UPDATE_SCHOOL_ZONE (state, payload) {
    state.filter.schools.zone = payload
  },
  UPDATE_PLOT_PRIMARY_SCHOOLS (state, payload) {
    state.filter.schools.primary.plot = payload
  },
  UPDATE_SCHOOL_PRIMARY_RATING (state, payload) {
    state.filter.schools.primary.rating = Number(payload)
  },
  UPDATE_SCHOOL_PRIMARY_ENG_RATING (state, payload) {
    state.filter.schools.primary.englishRating = Number(payload)
  },
  UPDATE_SCHOOL_PRIMARY_MATH_RATING (state, payload) {
    state.filter.schools.primary.mathsRating = Number(payload)
  },
  UPDATE_PLOT_SECONDARY_SCHOOLS (state, payload) {
    state.filter.schools.secondary.plot = payload
  },
  UPDATE_SCHOOL_SECONDARY_RATING (state, payload) {
    state.filter.schools.secondary.rating = Number(payload)
  },
  UPDATE_SCHOOL_SECONDARY_ENG_RATING (state, payload) {
    state.filter.schools.secondary.englishRating = Number(payload)
  },
  UPDATE_SCHOOL_SECONDARY_MATH_RATING (state, payload) {
    state.filter.schools.secondary.mathsRating = Number(payload)
  },
  ADD_GEOCODE_RESULTS (state, payload) {
    state.geocodeSuggestions = payload.map((f: any) => ({
      center: f.center,
      name: f.place_name,
      placeType: f.place_type
    }))
  }
}

export const actions: ActionTree<RootState, RootState> = {
  async loadListings ({ commit, state }) {
    const { filter, bounds } = state

    const payload = {
      listingType: 'Sale',
      pageSize: 200,
      propertyTypes: filter.properties.types,
      minBedrooms: filter.properties.rooms,
      minBathrooms: filter.properties.bathrooms,
      minCarspaces: filter.properties.carSpaces,
      maxPrice: filter.properties.maxPrice,
      locations: [{ state: 'VIC' }],
      geoWindow: { box: { topLeft: bounds.nw, bottomRight: bounds.se } }
    }

    const listings = await this.$axios.$post('https://api.domain.com.au/v1/listings/residential/_search', payload, { headers: {'X-Api-Key': process.env.domainApiKey } })
    commit('ADD_LISTINGS', listings)
  },

  async geocode ({ commit }, searchTerm) {
    if (searchTerm) {
      const params = {
        key: process.env.mapTilerSecret,
        bbox: '141.24847451171752,-39.624732230379166,148.68072548828115,-34.32302549095252',
        language: 'en'
      }

      const geocodeResults = await this.$axios.$get(`https://api.maptiler.com/geocoding/${searchTerm}.json`, { params })
      const schoolsResults = await this.$axios.$get(`//${process.env.domain}/schools/schools.json`)
      const mappedSchoolResults = schoolsResults.map((s: any) => ({
        center: [s.lon, s.lat],
        place_name: s.schoolName,
        place_type: 'school'
      })).filter((s: any) => s.place_name.toLowerCase().includes(searchTerm))

      commit('ADD_GEOCODE_RESULTS', [...geocodeResults.features, ...mappedSchoolResults])
    } else {
      commit('ADD_GEOCODE_RESULTS', [])
    }
  },

  setBoundingBox({ commit }, payload) {
    commit('UPDATE_BOUNDING_BOX', payload)
  }
}
