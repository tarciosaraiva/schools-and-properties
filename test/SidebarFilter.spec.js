import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import { state } from '@/store'
import SidebarFilter from '@/components/SidebarFilter.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('SidebarFilter', () => {
  let store
  let wrapper
  let propsData
  let mutations

  beforeEach(() => {
    propsData = { loadListingsFn: jest.fn() }
    mutations = {
      'UPDATE_PROPERTY_FILTER_TYPE': jest.fn(),
      'UPDATE_PROPERTY_FILTER_ROOMS': jest.fn(),
      'UPDATE_PROPERTY_FILTER_BATHROOMS': jest.fn(),
      'UPDATE_PROPERTY_FILTER_CAR_SPACES': jest.fn(),
      'UPDATE_PROPERTY_FILTER_MAX_PRICE': jest.fn(),
      'UPDATE_SCHOOL_EDUCATION_SECTOR': jest.fn(),
      'UPDATE_SCHOOL_RATING': jest.fn(),
      'UPDATE_SCHOOL_ENG_RATING': jest.fn(),
      'UPDATE_SCHOOL_MATH_RATING': jest.fn()
    }
    store = new Vuex.Store({ state, mutations })
    wrapper = shallowMount(SidebarFilter, { propsData, store, localVue })
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('property types default values matches state', () => {
    const houseInput = wrapper.find('#p_house_type')
    expect(houseInput.element.checked).toBeTruthy()

    const townhouseInput = wrapper.find('#p_townhouse_type')
    expect(townhouseInput.element.checked).toBeTruthy()

    const unitInput = wrapper.find('#p_unit_type')
    expect(unitInput.element.checked).toBeFalsy()
  })

  test('rooms default value match state', () => {
    const input = wrapper.find('#p_rooms')
    expect(Number(input.element.value)).toBe(store.state.filter.properties.rooms)
  })

  test('bathrooms default value match state', () => {
    const input = wrapper.find('#p_bathrooms')
    expect(Number(input.element.value)).toBe(store.state.filter.properties.bathrooms)
  })

  test('car spaces default value match state', () => {
    const input = wrapper.find('#p_car_spaces')
    expect(Number(input.element.value)).toBe(store.state.filter.properties.carSpaces)
  })

  test('max price default value match state', () => {
    const input = wrapper.find('#p_max_price')
    expect(Number(input.element.value)).toBe(store.state.filter.properties.maxPrice)
  })

  test('education sector default value matches state', () => {
    const inputs = wrapper.findAll('#s_edu_sector')
    const checkedElement = inputs.filter(i => i.element.checked)
    expect(checkedElement).toHaveLength(1)
    expect(checkedElement.at(0).element.value).toBe(store.state.filter.schools.educationSector)
  })

  test('overall rating default value match state', () => {
    const input = wrapper.find('#s_rating')
    expect(Number(input.element.value)).toBe(store.state.filter.schools.rating)
  })

  test('english rating default value match state', () => {
    const input = wrapper.find('#s_english_rating')
    expect(Number(input.element.value)).toBe(store.state.filter.schools.englishRating)
  })

  test('maths rating default value match state', () => {
    const input = wrapper.find('#s_maths_rating')
    expect(Number(input.element.value)).toBe(store.state.filter.schools.mathsRating)
  })

  test('should invoke passed "loadListingFn" when applying filter', () => {
    const button = wrapper.find('button')
    button.trigger('click')
    expect(propsData.loadListingsFn).toHaveBeenCalled()
  })

  test.each([
    ['UPDATE_PROPERTY_FILTER_ROOMS', '#p_rooms', '2'],
    ['UPDATE_PROPERTY_FILTER_BATHROOMS', '#p_bathrooms', '3'],
    ['UPDATE_PROPERTY_FILTER_CAR_SPACES', '#p_car_spaces', '2'],
    ['UPDATE_PROPERTY_FILTER_MAX_PRICE', '#p_max_price', '500000'],
    ['UPDATE_SCHOOL_RATING', '#s_rating', '99'],
    ['UPDATE_SCHOOL_ENG_RATING', '#s_english_rating', '2'],
    ['UPDATE_SCHOOL_MATH_RATING', '#s_maths_rating', '3'],
  ])(
    'should have committed mutation %p to the store when rating changes',
    (mutation, elementId, value) => {
      const input = wrapper.find(elementId)
      input.element.value = value
      input.trigger('change')

      expect(mutations[mutation]).toHaveBeenCalledWith(store.state, value)
    }
  )
})
