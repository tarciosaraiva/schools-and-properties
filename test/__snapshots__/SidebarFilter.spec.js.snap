// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarFilter renders correctly with props and store 1`] = `
<div class="filter-container">
  <fieldset>
    <legend>Property filter</legend>
    <div class="row">
      <div class="col-sm">
        <p>Property type</p>
      </div>
    </div>
    <div class="row responsive-label">
      <div class="col-sm-1">
        <input id="p_house_type" type="checkbox" value="House">
      </div>
      <div class="col-sm">
        <label for="p_house_type">House</label>
      </div>
      <div class="col-sm-1">
        <input id="p_townhouse_type" type="checkbox" value="Townhouse">
      </div>
      <div class="col-sm">
        <label for="p_townhouse_type"> Townhouse </label>
      </div>
      <div class="col-sm-1">
        <input id="p_unit_type" type="checkbox" value="ApartmentUnitFlat">
      </div>
      <div class="col-sm">
        <label for="p_unit_type"> Unit </label>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="p_rooms">Rooms</label>
          <select id="p_rooms">
            <option value="1">1+</option>
            <option value="2">2+</option>
            <option value="3">3+</option>
            <option value="4">4+</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="p_bathrooms">Bathrooms</label>
          <select id="p_bathrooms">
            <option value="1">1+</option>
            <option value="2">2+</option>
            <option value="3">3+</option>
            <option value="4">4+</option>
          </select>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="p_car_spaces">Car spaces</label>
          <select id="p_car_spaces">
            <option value="1">1+</option>
            <option value="2">2+</option>
            <option value="3">3+</option>
            <option value="4">4+</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="p_max_price">Max price</label>
          <select id="p_max_price">
            <option value="500000">500 K</option>
            <option value="600000">600 K</option>
            <option value="700000">700 K</option>
            <option value="800000">800 K</option>
            <option value="900000">900 K</option>
            <option value="1000000">1 M</option>
          </select>
        </div>
      </div>
    </div>
    <div class="button-group">
      <button class="primary">
        <span class="icon-search inverse"></span> Find matching properties
      </button>
    </div>
  </fieldset>
  <fieldset>
    <legend>School filter</legend>
    <div class="row">
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_edu_sector">Education sector</label>
          <select id="s_edu_sector">
            <option value="all">All</option>
            <option value="Government">Government</option>
            <option value="Non-Government">Non Government</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_school_zones">School zones</label>
          <select id="s_school_zones">
            <option value="none">None</option>
            <option value="primary">Primary</option>
            <option value="p7">Secondary P7</option>
            <option value="p8">Secondary P8</option>
            <option value="p9">Secondary P9</option>
            <option value="p10">Secondary P10</option>
            <option value="p11">Secondary P11</option>
            <option value="p12">Secondary P12</option>
          </select>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <p>Primary school ratings</p>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_prim_rating">Overall</label>
          <select id="s_prim_rating">
            <option value="100">= 100</option>
            <option value="99">&gt;= 99</option>
            <option value="98">&gt;= 98</option>
            <option value="97">&gt;= 97</option>
            <option value="96">&gt;= 96</option>
            <option value="95">&gt;= 95</option>
            <option value="94">&gt;= 94</option>
            <option value="93">&gt;= 93</option>
            <option value="92">&gt;= 92</option>
            <option value="91">&gt;= 91</option>
            <option value="90">&gt;= 90</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_prim_english_rating">English</label>
          <select id="s_prim_english_rating">
            <option value="5">= 5</option>
            <option value="4">&gt;= 4</option>
            <option value="3">&gt;= 3</option>
            <option value="2">&gt;= 2</option>
            <option value="1">&gt;= 1</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_prim_maths_rating">Maths</label>
          <select id="s_prim_maths_rating">
            <option value="5">= 5</option>
            <option value="4">&gt;= 4</option>
            <option value="3">&gt;= 3</option>
            <option value="2">&gt;= 2</option>
            <option value="1">&gt;= 1</option>
          </select>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <p>Secondary school ratings</p>
      </div>
    </div>
    <div class="row">
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_sec_rating">Overall</label>
          <select id="s_sec_rating">
            <option value="100">= 100</option>
            <option value="99">&gt;= 99</option>
            <option value="98">&gt;= 98</option>
            <option value="97">&gt;= 97</option>
            <option value="96">&gt;= 96</option>
            <option value="95">&gt;= 95</option>
            <option value="94">&gt;= 94</option>
            <option value="93">&gt;= 93</option>
            <option value="92">&gt;= 92</option>
            <option value="91">&gt;= 91</option>
            <option value="90">&gt;= 90</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_sec_english_rating">English</label>
          <select id="s_sec_english_rating">
            <option value="5">= 5</option>
            <option value="4">&gt;= 4</option>
            <option value="3">&gt;= 3</option>
            <option value="2">&gt;= 2</option>
            <option value="1">&gt;= 1</option>
          </select>
        </div>
      </div>
      <div class="col-sm">
        <div class="input-group vertical">
          <label for="s_sec_maths_rating">Maths</label>
          <select id="s_sec_maths_rating">
            <option value="5">= 5</option>
            <option value="4">&gt;= 4</option>
            <option value="3">&gt;= 3</option>
            <option value="2">&gt;= 2</option>
            <option value="1">&gt;= 1</option>
          </select>
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
